lib/timers.js

```diff
@@ -635,36 +639,39 @@ ImmediateList.prototype.remove = function(item) {
 };
 
 // Create a single linked list instance only once at startup
-var immediateQueue = new ImmediateList();
+const immediateQueue = new ImmediateList();
+
+// If an error is encountered during execution of immediateQueue,
+// this queue will store all remaining Immediates that need to run
+// upon resolution of all error handling (if process is still alive).
+const outstandingQueue = new ImmediateList();
 
 
 function processImmediate() {
-  var immediate = immediateQueue.head;
-  var tail = immediateQueue.tail;
-  var domain;
+  const queue = outstandingQueue.head !== null ?
+    outstandingQueue : immediateQueue;
+  var immediate = queue.head;
+  var tail = queue.tail;
 
   // Clear the linked list early in case new `setImmediate()` calls occur while
   // immediate callbacks are executed
-  immediateQueue.head = immediateQueue.tail = null;
+  queue.head = queue.tail = null;
 
   while (immediate !== null) {
-    domain = immediate.domain;
-
     if (!immediate._onImmediate) {
       immediate = immediate._idleNext;
       continue;
     }
 
-    if (domain)
-      domain.enter();
```

Comment:
Just making sure: These are a drive-by fix and not related to the exception handling itself, because domains use async hooks now anyway. Right?

Reply:
Correct.

Reply:
I'm actually thinking that this should be a separate PR because this change could be back-ported but the domain portion probably can't.